modules/images/image-loading-optimization/rest-api.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * REST API integration for the module.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Register endpoint for storage of metrics.
+ */
+function image_loading_optimization_register_endpoint() {
+
+	$dom_rect_schema = array(
+		'type'       => 'object',
+		'properties' => array(
+			'width'  => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			'height' => array(
+				'type'    => 'number',
+				'minimum' => 0,
+			),
+			// TODO: There are other properties to define if we need them: x, y, top, right, bottom, left.
+		),
+	);
+
+	register_rest_route(
+		'perflab/v1',
+		'/image-loading-optimization/metrics-storage',
+		array(
+			'methods'             => 'POST',
+			'callback'            => 'image_loading_optimization_handle_rest_request',
+			'permission_callback' => '__return_true', // Needs to be available to unauthenticated visitors.
```

Comment:
If it's going to be public, this probably needs some other kind of protection to prevent being flooded with entries.

Years ago I built something similar where multiple requests were blocked by storing the IP address in a transient. Maybe it's useful here:

https://github.com/wearerequired/rest-likes/blob/8c522be82ca468622d287c7a8539ea1dc72e45e9/classes/Controller.php#L251-L253
https://github.com/wearerequired/rest-likes/blob/8c522be82ca468622d287c7a8539ea1dc72e45e9/classes/Controller.php#L282-L353

Reply:
Thank you! Yes, I was also intending to do something similar following the same pattern as WP comments.